src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,27 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws ParseException {
+        SimpleDateFormat df = new SimpleDateFormat("dd.MM.yyyy");
+        String report = "Report for period " + dateFrom + " - " + dateTo;
+
+        for (int i = 0; i < names.length; i++) {
+            int salary = 0;
+            for (int k = 0; k < data.length; k++) {
+                String[] workInfo = data[k].split(" ");
+                if (names[i].equals(workInfo[1])) {
+                    if (!df.parse(workInfo[0]).before(df.parse(dateFrom))
+                            && !df.parse(workInfo[0]).after(df.parse(dateTo))) {
```

Comment:
you parse date from data line twice let's create a variable for it and use it here